front/components/data_source/WebsiteConfiguration.tsx

```diff
@@ -0,0 +1,365 @@
+import {
+  ContentMessage,
+  DropdownMenu,
+  Input,
+  Page,
+  RadioButton,
+} from "@dust-tt/sparkle";
+import type {
+  CrawlingFrequency,
+  DataSourceType,
+  DepthOption,
+  SubscriptionType,
+  UpdateConnectorRequestBodySchema,
+  WebCrawlerConfigurationType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import type { APIError } from "@dust-tt/types";
+import {
+  CrawlingFrequencies,
+  DepthOptions,
+  WEBCRAWLER_MAX_PAGES,
+} from "@dust-tt/types";
+import type * as t from "io-ts";
+import { useRouter } from "next/router";
+import { useCallback, useEffect, useState } from "react";
+
+import AppLayout from "@app/components/sparkle/AppLayout";
+import { AppLayoutSimpleSaveCancelTitle } from "@app/components/sparkle/AppLayoutTitle";
+import { subNavigationBuild } from "@app/components/sparkle/navigation";
+import { urlToDataSourceName } from "@app/lib/webcrawler";
+import type { PostManagedDataSourceRequestBodySchema } from "@app/pages/api/w/[wId]/data_sources/managed";
+
+export default function WebsiteConfiguration({
+  owner,
+  subscription,
+  dataSources,
+  dataSource,
+  gaTrackingId,
+  webCrawlerConfiguration,
+}: {
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  dataSources: DataSourceType[];
+  webCrawlerConfiguration: WebCrawlerConfigurationType | null;
+  dataSource: DataSourceType | null;
+  gaTrackingId: string;
+}) {
+  const [isSaving, setIsSaving] = useState(false);
+  const [isEdited, setIsEdited] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+
+  const [dataSourceNameError, setDataSourceNameError] = useState("");
+  const [dataSourceUrl, setDataSourceUrl] = useState(
+    webCrawlerConfiguration?.url || ""
+  );
+  const [maxPages, setMaxPages] = useState<number | null>(
+    webCrawlerConfiguration?.maxPageToCrawl || 50
```

Comment:
If 50 is the maximum, can we use the Macro?

Reply:
which macro?

Reply:
Nevermind, just got it. I fixed the code to use it at the parsing site.